File: _정답코드/1주차/06_한글_숫자_읽기/solution.java
import java.util.Map;
import java.util.LinkedHashMap;

class Solution {
    public String solution(String hangul_numbers) {
        Map<String, String> map = new LinkedHashMap<>();
        String[] hangul = {"공", "하나", "둘", "셋", "넷", "다섯", "여섯", "일곱", "여덟", "아홉",
                           "영", "일", "이", "삼", "사", "오", "육", "칠", "팔", "구"};
        String[] digit = {"0", "1", "2", "3", "4", "5", "6", "7", "8", "9",
                          "0", "1", "2", "3", "4", "5", "6", "7", "8", "9"};
        for (int i = 0; i < hangul.length; i++) map.put(hangul[i], digit[i]);

        String result = hangul_numbers;
        for (Map.Entry<String, String> entry : map.entrySet()) {
            result = result.replace(entry.getKey(), entry.getValue());
        }
        return result;
    }
}

File: _정답코드/1주차/06_한글_숫자_읽기/solution.py
def solution(hangul_numbers):
    hangul_mapping = {
        "공": "0", "하나": "1", "둘": "2", "셋": "3", "넷": "4",
        "다섯": "5", "여섯": "6", "일곱": "7", "여덟": "8", "아홉": "9",
        "영": "0", "일": "1", "이": "2", "삼": "3", "사": "4",
        "오": "5", "육": "6", "칠": "7", "팔": "8", "구": "9"
    }
    translate = hangul_numbers
    for hangul, digit in hangul_mapping.items():
        translate = translate.replace(hangul, digit)
    return translate